src/utils/inline-styles.js

```diff
@@ -24,11 +23,10 @@ export default class InlineStyles extends LitElement {
     // (allowed by the spec).
     const referenceCounts =
       crossTagReferenceCounts[this.tagName] || new WeakMap();
-    // one more reference, caused by this instance (undefined | 0 === 0)
-    const referenceCount = (referenceCounts.get(root) | 0) + 1;
-    // persist reference-count metadata
-    referenceCounts.set(root, referenceCount);
-    crossTagReferenceCounts[this.tagName] = referenceCounts;
+    // one more reference, caused by this instance
+    const info = referenceCounts.get(root) || {};
+    const referenceCount = (info.referenceCount || 0) + 1;
+    info.referenceCount = referenceCount;
```

Comment:
At line 28 you add +1 to `info.referenceCount` and save it into a variable. And here you save this variable back to info.referenceCount? I would say an extra variable is not necessary here. It is better readable without it.

Reply:
You forgot line 31, where thanks to the variable I can get rid of an unnecessary `this` reference. So, overall, I think my solution is preferrable.